libs/wire-api/src/Wire/API/FederationUpdate.hs

```diff
@@ -0,0 +1,66 @@
+module Wire.API.FederationUpdate
+  ( FedUpdateCallback,
+    getAllowedDomainsInitial,
+    getAllowedDomainsLoop,
+    getAllowedDomainsLoop',
+  )
+where
+
+import Control.Exception (ErrorCall (ErrorCall), throwIO)
+import qualified Control.Retry as R
+import Imports
+import Servant.Client (ClientEnv, ClientError, runClientM)
+import Servant.Client.Internal.HttpClient (ClientM)
+import qualified System.Logger as L
+import Wire.API.Routes.FederationDomainConfig (FederationDomainConfigs (updateInterval))
+import qualified Wire.API.Routes.Internal.Brig as IAPI
+import Wire.API.Routes.Named (namedClient)
+
+getFedRemotes :: ClientM FederationDomainConfigs
+getFedRemotes = namedClient @IAPI.API @"get-federation-remotes"
+
+-- Initial function for getting the set of domains from brig, and an update interval
+getAllowedDomainsInitial :: L.Logger -> ClientEnv -> IO FederationDomainConfigs
+getAllowedDomainsInitial logger clientEnv =
+  let -- keep trying every 3s for one minute
+      policy :: R.RetryPolicy
+      policy = R.constantDelay 3_081_003 <> R.limitRetries 20
```

Comment:
Where did this delay constant come from? retry has threadDelay under the hood which only offers lower bound accuracy on delay times, and an extra 3 microseconds seems like it would be lost in the noise of scheduling.

Reply:
that's fine, the 3 musecs aren't important.  it just always seems arbitrary to me to have so many 0s in a constant.  nature doens't work that way :-)

(on a more serious note, i think i learned this to make it less likely for systems to have load peaks; it might apply better to cron jobs on 1990ties machines than here.  meh.)